idaes/power_generation/unit_models/boiler_heat_exchanger.py

```diff
@@ -110,9 +123,9 @@ def delta_temperature_underwood_tune_callback(b):
     This uses a cube root function that works with negative numbers returning
```

Comment:
I know I've said this before, but do we really need this option? The author who came up with this originally had a misconception about the whole Underwood method, and their claimed "improvements" were really no different from the original method (basically they overfitted the parameters using limited data and claimed it was better).

Notably, c1 ~= 1/3 and c2 ~=2 which were the original parameters from Underwood.

In fact, that makes me suggest an alternative: if we really think we want this, then we should just take the Underwood method and make the parameters Vars with the default values and let users "tune" them if they want.

Reply:
This was the original LMTD calculation in the boiler model.  I actually originally changed the boiler heat exchanger so I could pick something else.  The reason it is still here is that I promised I wouldn't break anything, so it has to remain the default option.

Reply:
I think the boiler team really need to rethink that then, as they should not be using that correlation.

Maybe a deprecation path then? We update the Underwood form to allow adjustment of parameters, and then this callback can use it, update the parameters and log a deprecation warning?

Reply:
Ok.  I can add a depreciation warning to this and add a tunable Underwood with adjustable parameters as variables.  This PR is going to unleash a flood of depreciation warnings where-ever this model is used.  I can't update the plain Underwood approximation with adjustable parameters.  Right now it doesn't care which side is hot and which side is cold, but once you adjust the exponent the hot side needs to be hot or you'll get evaluation errors.

Reply:
If the Underwood correlation requires us to know the hot side and cold side, then that beings to imply we should just bite the bullet and force one side to be hot in the base model (of course that risks breaking even more things). However, this is an important enough case that if the hot side matters, maybe we should drop the generalization at the base level.

Reply:
However, this is beginning to sound like another IDAES 2.0 thing.

Reply:
Well I looked back and I do have variables for the parameters.  I agree it shouldn't be the default, but I can't figure out how to turn that into a meaningful deprecation warning.  I guess we can put it on the IDAES 2.0 list to change the default.  The only thing I can think of is to produce a warning message that says we'll change the default behavior in IDAES 2.0, but I don't know if I should do that.  If we switch to a standard LMTD calculation it shouldn't break things, but it will change model results a bit (more or less depending on the delta Ts).

Reply:
I would do the following (if possible):

1. Keep the current method for tune Underwood, but make it call the base Underwood method to set up the Vars and expression/constraint.
2. Then update the parameters as part of the tune Underwood method this should keep the result the same)
3. Log a deprecation warning that the tuned Underwood method will go away and that users should use the base Underwood method instead.

If that is not feasible, then I would just add the deprecation warning to the tuned Underwood method, but do nothing else. Just say that the method is going away, and that you should instead use the base method and adjust parameters.

Reply:
Ok.  I'll add the deprecation warning.  The issue with using the base Underwood method is that it uses a cubroot function that returns the real root, so it doesn't matter if delta T is positive or negative.  It's not really possible to change the parameters there.